lib/core/src/lib/auth/oidc/auth.service.ts

```diff
@@ -0,0 +1,60 @@
+/*!
+ * @license
+ * Copyright 2019 Alfresco Software, Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { TokenResponse } from 'angular-oauth2-oidc';
+import { Observable } from 'rxjs';
+
+/**
+ * Provide authentication/authorization through OAuth2/OIDC protocol.
+ */
+export abstract class AuthService {
+  /** Subscribe to whether the user has valid Id/Access tokens.  */
+  abstract authenticated$: Observable<boolean>;
+
+  /** Get whether the user has valid Id/Access tokens. */
+  abstract authenticated: boolean;
+
+  /** Subscribe to errors reaching the IdP. */
+  abstract idpUnreachable$: Observable<Error>;
+
+  /** Get user profile, if authenticated. */
+  abstract getUserProfile<T>(): Promise<T>;
```

Comment:
I'm not sure whether it is a good idea to mix user profile related things with authentication.

Reply:
It can be removed, the main intention was to provide an method to access informations about user, stored inside IDP provider.